go/encoding/basex/encoding.go

```diff
@@ -178,8 +207,13 @@ func (enc *Encoding) decodeBlock(dst []byte, src []byte, baseOffset int, strict
 		return 0, 0, ErrInvalidEncodingLength
 	}
 
-	raw := res.Bytes()
 	paddedLen := enc.DecodedLen(numGoodChars)
+
+	// Compute the corresponding right shift, to move the number
+	// over to its natural base.
+	res = res.Rsh(res, enc.extraBits(paddedLen, numGoodChars))
+
+	raw := res.Bytes()
```

Comment:
This and `SetBytes` above both rely on the fact that Go's `big` library is big-endian by default. Can we comment that? That might be one of the things that someone reimplementing our format gets wrong and loses a lot of time on.